crates/editor/src/element.rs

```diff
@@ -3145,92 +3160,117 @@ impl EditorElement {
             return Arc::default();
         }
 
-        let (newest_selection_head, is_relative) = self.editor.update(cx, |editor, cx| {
-            let newest_selection_head = newest_selection_head.unwrap_or_else(|| {
-                let newest = editor.selections.newest::<Point>(cx);
-                SelectionLayout::new(
-                    newest,
-                    editor.selections.line_mode(),
-                    editor.cursor_shape,
-                    &snapshot.display_snapshot,
-                    true,
-                    true,
-                    None,
+        let (newest_selection_head, is_relative, use_relative_for_wrapped_lines) =
+            self.editor.update(cx, |editor, cx| {
+                let newest_selection_head = newest_selection_head.unwrap_or_else(|| {
+                    let newest = editor.selections.newest::<Point>(cx);
+                    SelectionLayout::new(
+                        newest,
+                        editor.selections.line_mode(),
+                        editor.cursor_shape,
+                        &snapshot.display_snapshot,
+                        true,
+                        true,
+                        None,
+                    )
+                    .head
+                });
+                let is_relative = editor.should_use_relative_line_numbers(cx);
+                let use_relative_for_wrapped_lines = is_relative
+                    && EditorSettings::get_global(cx).relative_line_numbers_for_wrapped_lines;
```

Comment:
We probably only want this true if relative lines is true AND the setting is true. I wonder if this would be better served by an enum -- `relative_line_numbers: None | Logical | Wrapped` or something? Of course, the `should_use_relative_line_numbers` probably also checks for the current state of `vim.toggle_relative_line_numbers`, so maybe not?